shared/src/commonMain/kotlin/org/hyperskill/app/step/presentation/StepActionDispatcher.kt

```diff
@@ -164,4 +180,58 @@ internal class StepActionDispatcher(
             }
         }
     }
+
+    private fun handleCreateShareLink(
+        stepRoute: StepRoute,
+        onNewMessage: (Message) -> Unit
+    ) {
+        onNewMessage(
+            InternalMessage.ShareLinkReady(
+                urlBuilder.build(HyperskillUrlPath.Step(stepRoute)).toString()
+            )
+        )
+    }
+
+    private suspend fun handleGetMagicLink(
+        path: HyperskillUrlPath,
+        onNewMessage: (Message) -> Unit
+    ) {
+        magicLinksInteractor
+            .createMagicLink(path.path)
+            .fold(
+                onSuccess = {
+                    InternalMessage.GetMagicLinkReceiveSuccess(it.url)
+                },
+                onFailure = { InternalMessage.GetMagicLinkReceiveFailure }
+            )
+            .let(onNewMessage)
+    }
+
+    private suspend fun handleSkipStep(
+        stepId: Long,
+        onNewMessage: (Message) -> Unit
+    ) {
+        sentryInteractor.withTransaction(
+            transaction = HyperskillSentryTransactionBuilder.buildSkipStep(),
+            onError = { InternalMessage.StepSkipFailed }
+        ) {
+            stepInteractor.skipStep(stepId)
+            InternalMessage.StepSkipSuccess
+        }.let(onNewMessage)
+    }
+
+    private suspend fun handleFetchNextLearningActivityStep(
+        onNewMessage: (Message) -> Unit
+    ) {
+        sentryInteractor.withTransaction(
+            transaction = HyperskillSentryTransactionBuilder.buildSkipStepNextLearningActivityLoading(),
+            onError = { InternalMessage.FetchNextLearningActivityError }
+        ) {
+            val nextLearningActivity =
+                learningActivityRepository.getNextLearningActivity(
```

Comment:
@ivan-magda 
`NextLearningActivityStateRepository` doesn't pass Learning activity types into the request, and the back-end can return the stage instead of the step learning activity. This is why I use the `LearningActivityRepository` directly, as it is used in the `FirstProblemOnboardingActionDispatcher`.
https://github.com/hyperskill/mobile-app/blob/830503ab7d878c1476ad9208546f709b9e18e791/shared/src/commonMain/kotlin/org/hyperskill/app/first_problem_onboarding/presentation/FirstProblemOnboardingActionDispatcher.kt#L48-L67

Reply:
Okay, I found, that in the `StepCompletionActionDispacther` the `NextLearningActivityStateRepository ` is used.
I will use and filter the same way, as in the `StepCompletionReducer`.
https://github.com/hyperskill/mobile-app/blob/830503ab7d878c1476ad9208546f709b9e18e791/shared/src/commonMain/kotlin/org/hyperskill/app/step_completion/presentation/StepCompletionReducer.kt#L216-248